lib/spack/llnl/util/tty/__init__.py

```diff
@@ -3,13 +3,16 @@
 #
 # SPDX-License-Identifier: (Apache-2.0 OR MIT)
 
+from __future__ import unicode_literals
```

Comment:
> I'm curious whether #10186 solves the root problem, though -- if it does, maybe we don't need this import everywhere?

Unfortunately rebasing on `develop` and checking if #10186 fixed this issue was my first try and it didn't work :sob: Using 3b8b13809e4f9e8d7456c4d2e27d7ff069c8e44e:
```console
$ python2.6 bin/spack -d python
==> Reading config file /home/mculpo/PycharmProjects/spack/etc/spack/defaults/modules.yaml
==> Reading config file /home/mculpo/PycharmProjects/spack/etc/spack/defaults/linux/modules.yaml
==> Reading config file /home/mculpo/PycharmProjects/spack/etc/spack/defaults/config.yaml
==> Reading config file /home/mculpo/.spack/linux/config.yaml
Spack version 0.12.0
Python 2.6.9, Linux x86_64
>>> import llnl.util.tty as tty
>>> tty.info(u'Some unicode string')
==> Some unicode string
>>> tty.info(u'Some unicode string\u2018')
Traceback (most recent call last):
  File "<console>", line 1, in <module>
  File "/home/mculpo/PycharmProjects/spack/lib/spack/llnl/util/tty/__init__.py", line 91, in info
    cprint("@%s{%s==>} %s" % (format, st_text, cescape(str(message))),
UnicodeEncodeError: 'ascii' codec can't encode character u'\u2018' in position 19: ordinal not in range(128)

$ python2.7 bin/spack -d python
==> Reading config file /home/mculpo/PycharmProjects/spack/etc/spack/defaults/modules.yaml
==> Reading config file /home/mculpo/PycharmProjects/spack/etc/spack/defaults/linux/modules.yaml
==> Reading config file /home/mculpo/PycharmProjects/spack/etc/spack/defaults/config.yaml
==> Reading config file /home/mculpo/.spack/linux/config.yaml
Spack version 0.12.0
Python 2.7.15rc1, Linux x86_64
>>> import llnl.util.tty as tty
>>> tty.info(u'Some unicode string')
==> Some unicode string
>>> tty.info(u'Some unicode string\u2018')
Traceback (most recent call last):
  File "<console>", line 1, in <module>
  File "/home/mculpo/PycharmProjects/spack/lib/spack/llnl/util/tty/__init__.py", line 91, in info
    cprint("@%s{%s==>} %s" % (format, st_text, cescape(str(message))),
UnicodeEncodeError: 'ascii' codec can't encode character u'\u2018' in position 19: ordinal not in range(128)
```
one can see that `tty` fails if we try to print some non-ascii character in Python 2. Here I have been able to solve the issue for Python 2.7 but for Python 2.6 I found no solution - the error seems in the implementation of `sys.stdout.write`, see https://github.com/spack/spack/pull/10190#issuecomment-449962219. I didn't dig deeper, but given that Python 2.6 suport has been discontinued a long time ago it might be a bug in the implementation of `sys.stdout`.

Reply:
ok I think it's a reasonable fix, then.